File: src/test_donkey_kong.cpp
#include "NES.h"
#include "Renderer.h"
#include <string>

int
main() {
    std::string filepath = "../roms/donkey_kong/donkey_kong.nes";

    NES nes = NES(filepath);
    nes.ppu.renderer.render_type = RenderType::FILE;
    // Running for 5 frames is enough to get to the title screen,
    // which should be good enough as a test so far.
    while (nes.ppu.frames < 22) {
        nes.next_frame();
    }
    return 0;
}

File: py/tests/test_donkey_kong.py
from pathlib import Path

import pytest

from nes import NES


def test_donkey_kong():
    rom_path = Path(__file__).parent.joinpath(
        '../roms/donkey_kong/donkey_kong.nes'
    )

    nes = NES(rom_path)
    # Running for 5 frames is enough to get to the title screen,
    # which should be good enough as a test so far.
    while nes.ppu._frames < 5:
        nes.run()
